src/classes/CRLP_Operation_SVC.cls

```diff
@@ -154,39 +154,53 @@ public class CRLP_Operation_SVC {
      * @return Amount (converted to the currency of the summary record)
      */
     private static Decimal getAmountValue(CRLP_Rollup rollup, Opportunity oppSObject, SObject amtSObject, Date dateValue) {
-        Decimal amountValue = null;
+        Decimal amount = null;
         String amtField = rollup.amountFieldName;
 
         // Hardcoding this performance tweak similar to the Opportunity.CloseDate field above
         if (rollup.AmountObject == oppObjectName && amtField == 'Amount') {
-            amountValue = oppSObject.Amount;
-        }
-        if (amountValue == null && amtField != null && amtSObject.get(amtField) != null) {
-            amountValue = (Decimal) amtSObject.get(amtField);
+            amount = oppSObject.Amount;
         }
 
-        // If multi currency is enabled in the organization, get the "From" CurrencyCode to use for conversion
-        String fromCurrCode = UserInfo.getDefaultCurrency(); // default just in case there's an issue getting the detail record currency
-        if (UTIL_Currency.getInstance().isMultiCurrencyOrganization()) {
+        if (amount == null && amtField != null) {
             try {
-                // Try to get the CurrencyIsoCode
-                Object currObj = amtSObject.get('CurrencyIsoCode');
-                if (currObj != null) {
-                    fromCurrCode = (String)currObj;
-                }
-            } catch (Exception ex) { }
+                amount = (Decimal) amtSObject.get(amtField);
+            } catch (Exception e) {
+                // The amount field is stale and it is not included in the rollup definition
```

Comment:
It's not that the Amount field is stale and it is actually specified on the Rollup Defintion, but it's not included in the Query because it was unnecessary due to the type of rollup. For example a Count rollup would not need to include an Amount field, but it's possible that the Amount field on the Rollup CMT has value in it pointing to a field that isn't in the query because it should be unused. Anyway, you can probably just remove this comment.

Reply:
No prob, I will remove the comment. However, here we have an Amount field populated and 
tried to be gotten from the SObject even though the Amount field is not applicable to the Rollup definition. Even though we do not need it for this work item, in general, should the Amount field be reset to null when the Rollup type/definition is changed?

I have not tested with multi-currency amount since its related code was just extracted. I will test now and let you know.

Reply:
Multi-currency test passed:
- Created Household and Contact and set their currency to CAD
- Created a donation for $5,000 USD
- Total amounts on Contact are converted correctly:
<img width="731" alt="Screen Shot 2019-07-02 at 5 19 31 PM" src="https://user-images.githubusercontent.com/28743286/60554732-ae7b2900-9ced-11e9-85cb-d19125e678e5.png">